astropy/io/fits/tests/test_convenience.py

```diff
@@ -153,3 +153,23 @@ def test_tabledump(self):
         # test with datafile
         fits.tabledump(filename, datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
+
+        # test with cdfile, hfile
+        """
+        test for https://github.com//astropy/astropy/issues/7015
+        """
+        fits.tabledump(filename, datafile = self.temp('test_tb.txt'), hfile = self.temp('hfile.txt'), cdfile = self.temp('cdfile.txt'), overwrite = True )
+        assert os.path.isfile(self.temp('hfile.txt'))
+        assert os.path.isfile(self.temp('cdfile.txt'))
+        sizeh =  os.path.getsize(self.temp('hfile.txt'))
+        sizecd =  os.path.getsize(self.temp('cdfile.txt'))
+        assert sizeh > 0
+        assert sizecd > 0
```

Comment:
Testing that the file has some bytes is rather useless. You could maybe test its content, but I think the best way to proceed is to check that the BinTableHDU created by `tableload` is similar to the original file (same header and data).